ta/gcov/ta_gcov.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2020 NXP
+ */
+
+#define STR_TRACE_USER_TA "GCOV"
+
+#include <tee_internal_api.h>
+#include <tee_internal_api_extensions.h>
+
+#include <gcov.h>
+#include <pta_gcov.h>
+#include "ta_gcov.h"
+
+TEE_Result TA_CreateEntryPoint(void)
+{
+	return TEE_SUCCESS;
+}
+
+void TA_DestroyEntryPoint(void)
+{
+}
+
+TEE_Result TA_OpenSessionEntryPoint(uint32_t param_types,
+		TEE_Param  params[4], void **sess_ctx)
+{
+	uint32_t exp_param_types = TEE_PARAM_TYPES(TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE);
+	if (param_types != exp_param_types) {
+		EMSG("Wrong param_types, exp %x, got %x", exp_param_types,
+		     param_types);
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	(void)&params;
+	(void)&sess_ctx;
+
+	return TEE_SUCCESS;
+}
+
+void TA_CloseSessionEntryPoint(void *sess_ctx)
+{
+	(void)&sess_ctx;
+}
+
+static TEE_Result get_version(uint32_t param_types, TEE_Param params[4])
+{
+	uint32_t exp_param_types = TEE_PARAM_TYPES(TEE_PARAM_TYPE_VALUE_OUTPUT,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE);
+
+	if (param_types != exp_param_types) {
+		EMSG("Wrong param_types, exp %x, got %x", exp_param_types,
+		     param_types);
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	return gcov_get_version(&params[0].value.a);
+}
+
+static TEE_Result dump_core(uint32_t param_types, TEE_Param params[4])
+{
+	TEE_Result res;
+	uint32_t exp_param_types = TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INPUT,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE);
+	uint32_t cRT = TEE_TIMEOUT_INFINITE;
+	TEE_UUID uuid = (TEE_UUID)PTA_GCOV_UUID;
+	TEE_TASessionHandle sess;
```

Comment:
prefer always an init vlaue.  (`ret` above also, and those below)

Reply:
ok